gulpTasks/linkGitHooks.js

```diff
@@ -11,30 +10,31 @@ const symlink = Q.denodeify(fs.ensureSymlink);
 const copy = Q.denodeify(fs.copy);
 const readdir = Q.denodeify(fs.readdir);
 
-gulp.task('linkGitHooks', function() {
-  let noSuchFile = -2;
-  let gitHooksDir = '.git/hooks/';
-  let gitHooksSourceDir = 'hooks/';
+const gitHooksDir = '.git/hooks/';
+const gitHooksSourceDir = 'hooks/';
 
-  readdir(gitHooksSourceDir).then((files) => {
-    _.each(files, filename => {
-      let symname = path.resolve(process.cwd(), gitHooksDir + filename);
-      let source = path.resolve(process.cwd(), gitHooksSourceDir + filename);
-      symlink(source, symname, 'file')
-          .catch(err => {
-            //Need to be admin on windows to create a symlink (╯°□°）╯︵ ┻━┻
-            console.log('Couldn\'t create a symlink, trying to copy...'.yellow);
-            copy(source, symname)
-                .done(() => {
-                  console.log('Hook successfully copied');
-                })
-                .catch(err => {
-                  console.log(colors.red(err));
-                });
-          });
-    });
-  })
-  .catch(err => {
-    console.log(colors.red(err));
-  })
-});
\ No newline at end of file
+function linkGitHooks(cb) {
+  readdir(gitHooksSourceDir)
+    .then(files => _.each(files, filename => createSymLink(filename)))
+    .catch(err => console.log(colors.red(err)))
+    .finally(cb);
+}
+
+function createSymLink(filename) {
+  const symname = path.resolve(process.cwd(), gitHooksDir + filename);
+  const source = path.resolve(process.cwd(), gitHooksSourceDir + filename);
+
+  symlink(source, symname, 'file').catch(err => {
+    //Need to be admin on windows to create a symlink (╯°□°）╯︵ ┻━┻
+    console.log("Couldn't create a symlink, trying to copy...".yellow);
```

Comment:
While we're at it, perhaps it would be a good idea to change this error? Since the above comment implies this happens on Windows, perhaps adding this mention to that error would be a good idea?

Reply:
I think it is okay as-is. The comment suggests needing to be an admin on windows, but I do not know if there are other reasons why symlinks might not be created on other OSs. For example, if the error occurs on a unix system, then including windows in the logged message would not be very helpful.